.github/workflows/maven-publish.yml

```diff
@@ -0,0 +1,48 @@
+# This action script will make sure that project gets build and checked on every commit
+# and on pull request
+
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches-ignore: [master]
+  pull_request:
+    branches: [ master ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow is to build against java 8
+  build-ubuntu-java-8:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
```

Comment:
I think it's better to explicitly set the OS version, so it doesn't change unexpectedly

Reply:
Actually I kept it latest to make sure our code is not something OS specific, so if anything breaks we can have a look into the same

Reply:
I still don't agree. I think the decision when to upgrade the build host should be ours, not Githubs.